Packs/McAfeeNSM/Integrations/McAfeeNSMv2/McAfeeNSMv2.py

```diff
@@ -0,0 +1,1975 @@
+    Returns:
+        The correct response string.
+    """
+    split_str = response_str.upper().split()
+    if len(split_str) == 1:
+        return split_str[0]
+    else:
+        return '_'.join(split_str)
+
+
+def rule_object_type_cases(str_type: str, case: str) -> str:
+    """ Checks the rule_object_type params and returns the correct format of them.
+    Args:
+        str_type: str - The type string.
+        case: str - In what case should the letters be, upper or lower case.
+    Returns:
+        The correct rule_object_type string.
+    """
+    type_split = str_type.upper().replace('.', ' ').split()
+    if 'ENDPOINT' in type_split[0]:
+        r_type = f'HOST_IPV_{type_split[-1]}'
+    elif 'RANGE' in type_split[0]:
+        r_type = f'IPV_{type_split[-1]}_ADDRESS_RANGE'
+    else:
+        r_type = f'NETWORK_IPV_{type_split[-1]}'
+    if case == 'low':
+        return r_type.lower().replace('_', '')
+    return r_type
+
+
+def reverse_rule_object_type_cases(rule_type: str) -> str:
+    """ Checks the rule_object_type params that return from the API call and returns the matching string.
+    Args:
+        rule_type: str - The type string.
+    Returns:
+        The matching string.
+    """
+    number = '4' if ('4' in rule_type) else '6'
+    if 'HOST' in rule_type:
+        return f'Endpoint IP V.{number}'
+    elif 'ADDRESS_RANGE' in rule_type:
+        return f'Range IP V.{number}'
+    else:
+        return f'Network IP V.{number}'
+
+
+def check_source_and_destination(source_rule_object_id: Optional[int], source_rule_object_type: Optional[str],
+                                 destination_rule_object_id: Optional[int], destination_rule_object_type: Optional[str],
+                                 create_or_update: str):
+    """ Checks the source and destination objects.
+    Args:
+        source_rule_object_id: Optional[int] - Unique Rule Object ID.
+        source_rule_object_type: Optional[str] - Source / Destination Mode.
+        destination_rule_object_id: Optional[int] - Unique Rule Object ID.
+        destination_rule_object_type: Optional[str] - Source / Destination Mode.
+        create_or_update: str - From what function it was called.
+    Returns:
+        Throws exception .
+    """
+    if (source_rule_object_id and not source_rule_object_type and source_rule_object_id != -1) or (
+            not source_rule_object_id and source_rule_object_type):
+        raise Exception('If you provide at least one of the source fields, you must provide all of them.')
+    if (destination_rule_object_id and not destination_rule_object_type and destination_rule_object_id != -1) or \
+            (not destination_rule_object_id and destination_rule_object_type):
+        raise Exception('If you provide at least one of the destination fields, you must provide all of them.')
+    if create_or_update == 'create':
+        if source_rule_object_id == -1 and destination_rule_object_id == -1:
```

Comment:
Please check what happens if we don't provide source/destination to the create command without using the -1

Reply:
ok, so I think we can remove all the check for the -1
The error is clear and the check and everything related to it is very confusing and not necessary